packages/core/src/sync-store/postgres/store.ts

```diff
@@ -1047,16 +1021,39 @@ export class PostgresSyncStore implements SyncStore {
                   : { type: row.tx_type }),
         },
       } satisfies {
-        sourceId: string;
         chainId: number;
         log: Log;
         block: Block;
         transaction: Transaction;
       };
     });
 
+    stopClock = startClock();
+
     // Query for the checkpoint of the last event in the requested range (ignore the batch limit)
-    const lastCheckpointRows = await baseQuery
+    const lastCheckpointRows = await this.db
```

Comment:
It makes sense that these two queries don't need to be in a transaction, correct?

Should we consider running them at the same time?